src/js/mobile-menu.js

```diff
@@ -0,0 +1,69 @@
+/* global $ */
+(function () {
+
+  let menuSelector;
+  let menuTriggerSelector;
+  let $menuTrigger;
+  let menuTriggerClass;
+  let $menu;
+  let menuSlideoutClass;
+  let menuCloseSelector;
+  let $bodyOverlay;
+  let bodyOverlayClass;
+  let bodyNavOpenClass;
+
+
+  function closeMenu() {
+    $menu.removeClass(menuSlideoutClass);
+    $menuTrigger.removeClass(menuTriggerClass);
+    $bodyOverlay.removeClass(bodyOverlayClass);
+    $('body').removeClass(bodyNavOpenClass);
+  }
+
+  function toggleMenu() {
+    $menu.toggleClass(menuSlideoutClass);
+    $menuTrigger.toggleClass(menuTriggerClass);
+    $bodyOverlay.toggleClass(bodyOverlayClass);
+    $('body').toggleClass(bodyNavOpenClass);
+  }
+
+  function closeMobileMenuHandler() {
+    $(document).click((e) => {
+      const $target = $(e.target);
+
+      // Hide the mobile menu when anything else is clicked
+      if (!$target.closest(menuSelector).length && !$target.closest(menuTriggerSelector).length) {
```

Comment:
Something we may want to do, either in this PR or a later one, would be to check whether or not `$target` is a link, and abort closing the menu if true (so that the page doesn't jump when, say, a user clicks on a link in an article when they also happen to have the site nav open on desktop)

Reply:
While I was trying to figure out how to check whether the target was a link or not, I thought that we probably don't really want to close the nav menu if they click other items as well. And.. as I'm typing this out I'm thinking that we should probably have some sort of 'active' type style applied to the sections button (and maybe replace the hamburger menu icon with a close icon) when the desktop nav is open. But I think that can be done in a future PR :)

Back to the current issue, I ended up tackling this a bit differently than you described, so let me know if there's another scenario I haven't considered when solving it this way, but, I added a condition on the if statement so that the menu is only closed when clicking anything other than the close or 'Sections' menu button when the menu width is less than 977px (and I kept the condition of it also having to be larger than 200px). I logged the menu size right before the -md breakpoint to get that value.

See commit changes: dc19448